ts/components/TosWebviewComponent.tsx

```diff
@@ -1,94 +1,90 @@
+import { FlowType } from "../utils/analytics";
 import { NOTIFY_LINK_CLICK_SCRIPT } from "./ui/Markdown/script";
 import { WebViewMessage } from "./ui/Markdown/types";
-import { H2 } from "./core/typography/H2";
 
 type Props = {
   webViewSource: WebViewSource;
   handleLoadEnd: () => void;
   handleReload: () => void;
   onAcceptTos?: () => void;
-  onExit?: () => void;
   shouldRenderFooter?: boolean;
+  flow: FlowType;
 } & Pick<ViewProps, "testID">;
 
-const styles = StyleSheet.create({
-  errorContainer: {
-    padding: 20,
-    flex: 1,
-    justifyContent: "center",
-    alignItems: "center"
-  },
-  errorButtonsContainer: {
-    position: "absolute",
-    bottom: 30,
-    flex: 1,
-    flexDirection: "row"
-  }
-});
-
 const TosWebviewComponent: React.FunctionComponent<Props> = ({
   handleLoadEnd,
   handleReload,
   webViewSource,
   shouldRenderFooter,
-  onExit,
-  onAcceptTos
+  onAcceptTos,
+  flow
 }: Props) => {
   const [hasError, setHasError] = useState(false);
 
   const handleError = useCallback(() => {
     handleLoadEnd();
     setHasError(true);
-  }, [setHasError, handleLoadEnd]);
+    trackToSWebViewError(flow);
+  }, [handleLoadEnd, flow]);
 
   const handleRetry = useCallback(() => {
     handleReload();
     setHasError(false);
-  }, [setHasError, handleReload]);
+    trackToSWebViewErrorRetry(flow);
+  }, [handleReload, flow]);
 
   const renderError = () => (
-    <View style={styles.errorContainer} testID={"toSErrorContainerView"}>
-      <Image
-        source={brokenLinkImage}
-        resizeMode="contain"
-        testID={"toSErrorContainerImage"}
-      />
-      <View>
+    <>
+      <View
+        style={[
+          IOStyles.flex,
+          IOStyles.alignCenter,
+          IOStyles.horizontalContentPadding,
+          IOStyles.centerJustified
+        ]}
+        testID="toSErrorContainerView"
+      >
+        <Pictogram name="stopSecurity" />
         <VSpacer size={8} />
-        <H2 color="bluegrey" weight="Bold" testID={"toSErrorContainerTitle"}>
+        <H3 testID="toSErrorContainerTitle" style={{ textAlign: "center" }}>
           {I18n.t("onboarding.tos.error")}
-        </H2>
-      </View>
-
-      <View style={styles.errorButtonsContainer}>
-        <ButtonDefaultOpacity
-          onPress={handleRetry}
-          style={{ flex: 2 }}
-          block={true}
-          primary={true}
-          testID={"toSErrorContainerButton"}
-        >
-          <NBButtonText testID={"toSErrorContainerButtonText"}>
-            {I18n.t("global.buttons.retry")}
-          </NBButtonText>
-        </ButtonDefaultOpacity>
+        </H3>
       </View>
-    </View>
+      <FooterWithButtons
+        type="SingleButton"
+        sticky={true}
+        primary={{
+          type: "Solid",
+          buttonProps: {
+            onPress: () => handleRetry,
+            label: I18n.t("global.buttons.retry"),
+            accessibilityLabel: I18n.t("global.buttons.retry"),
```

Comment:
Since the translation is the same, what about putting it in a constant variable and use it instead? What do you think?

Reply:
No, for me it is not a good idea. This approach has not been used in the app, there is a practice of using labels and accessibility labels and not using variables to store the data.

Reply:
the DS library also added the functionality of not adding the accessibility label if it is equal to the label, so I prefer to use this approach here than the use of the variable.

Reply:
> Can you elaborate more? What "practice" are you referring to?

Reply:
> the DS library also added the functionality of not adding the accessibility label if it is equal to the label, so I prefer to use this approach here than the use of the variable.

Okay, that's good.

Reply:
In term of performance, calling a translation twice should be poor than calling it only once. That's the main reason why.  But as you are now calling it once (since you have removed the `accessibilityLabel` that falls back to `label` prop), in this case I agree with you that you can call it inline, because is only one call. What I was suggesting was using a constant in all those cases when you have to call the same translation (aka call `I18n.t("with the same key")`) multiple times.